src/test/java/hudson/security/docker/PlanetExpressTest.java

```diff
@@ -4,58 +4,69 @@
 import hudson.security.LDAPSecurityRealm;
 import hudson.tasks.MailAddressResolver;
 import hudson.util.Secret;
-import org.springframework.security.ldap.userdetails.LdapUserDetails;
-import org.jenkinsci.test.acceptance.docker.DockerContainer;
-import org.jenkinsci.test.acceptance.docker.DockerFixture;
-import org.jenkinsci.test.acceptance.docker.DockerRule;
-import org.junit.Rule;
-import org.junit.Test;
-import org.jvnet.hudson.test.JenkinsRule;
-
 import static org.hamcrest.core.StringContains.containsString;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
 import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RealJenkinsRule;
+import org.springframework.security.ldap.userdetails.LdapUserDetails;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
 
 /**
  * Tests the plugin when logging in to rroemhild/test-openldap
  */
 public class PlanetExpressTest {
 
+    static final String TEST_IMAGE = "rroemhild/test-openldap@sha256:b4e433bbcba1f17899d6bcb0a8e854bbe52c754faa4e785d0c27a2b55eb12cd8";
+    static final String DN = "dc=planetexpress,dc=com";
+    static final String MANAGER_DN = "cn=admin,dc=planetexpress,dc=com";
+    static final String MANAGER_SECRET = "GoodNewsEveryone";
+
+    @BeforeClass public static void requiresDocker() {
+        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
+    }
+
     @BeforeClass public static void linuxOnly() {
         assumeFalse("Windows CI builders now have Docker installed…but it does not support Linux images", Functions.isWindows() && System.getenv("JENKINS_URL") != null);
     }
 
+    @SuppressWarnings("rawtypes")
     @Rule
-    public DockerRule<PlanetExpress> docker = new DockerRule<>(PlanetExpress.class);
+    public GenericContainer container = new GenericContainer(TEST_IMAGE).withExposedPorts(389);
+
     @Rule
-    public JenkinsRule j = new JenkinsRule();
+    public RealJenkinsRule rr = new RealJenkinsRule();
```

Comment:
See https://github.com/jenkinsci/jenkins-test-harness/pull/280. Would have increased my confidence about class loading issues during #48 in particular.

Reply:
(I merged #77 to make this possible.)